src/main/java/com/example/onlinebookstore/exception/CustomGlobalExceptionHandler.java

```diff
@@ -33,6 +36,28 @@ protected ResponseEntity<Object> handleMethodArgumentNotValid(
         return new ResponseEntity<>(notValidResponse, headers, status);
     }
 
+    @ExceptionHandler(EntityNotFoundException.class)
+    protected ResponseEntity<Object> handleEntityNotFoundException(
+            EntityNotFoundException e
+    ) {
+        ArgumentNotValidResponse notValidResponse = new ArgumentNotValidResponse();
+        notValidResponse.setTimestamp(LocalDateTime.now());
+        notValidResponse.setStatus(HttpStatus.NOT_FOUND);
+        notValidResponse.setErrors(new String[]{e.getMessage()});
+        return new ResponseEntity<>(notValidResponse, HttpStatusCode.valueOf(NOT_FOUND_STATUS));
+    }
+
+    @ExceptionHandler(RegistrationException.class)
+    protected ResponseEntity<Object> handleRegistrationException(
+            RegistrationException e
+    ) {
+        ArgumentNotValidResponse notValidResponse = new ArgumentNotValidResponse();
+        notValidResponse.setTimestamp(LocalDateTime.now());
+        notValidResponse.setStatus(HttpStatus.NOT_FOUND);
```

Comment:
I'd recommend you throw 409 instead of 404 if an exception occurs during registration